src/kvstore/kvstore_dist.h

```diff
@@ -421,23 +466,198 @@ class KVStoreDist : public KVStoreLocal {
 
   /**
    * \brief cache all key partitions
+   *
+   * `ps_kv_` is used for row sparse
+   *
+   * `push_ps_kv_` and `pull_ps_kv_`, used for default type gradients, are same
+   * when there is no gradient compression
    */
   std::unordered_map<int, PSKV> ps_kv_;
+  std::unordered_map<int, PSKV> push_ps_kv_;
+  std::unordered_map<int, PSKV> pull_ps_kv_;
```

Comment:
Should the separate map be compress_ps_kv instead of splitting ps-kv to push/pull? We only need one map for normal kvstore without compression

Reply:
hmm, that might be better yes